mlflow/store/file_store.py

```diff
@@ -575,6 +598,19 @@ def log_param(self, run_id, param):
         make_containing_dirs(param_path)
         write_to(param_path, self._writeable_value(param.value))
 
+    def set_experiment_tag(self, experiment_id, tag):
+        """
+        Set a tag for the specified experiment
+
+        :param experiment_id: String ID of the experiment
+        :param tag: ExperimentRunTag instance to log
+        """
+        _validate_tag_name(tag.key)
+        # experiment validation happens in get_experiment_tag_path
```

Comment:
We aren't validating that the experiment is active. The following code sets a tag on a deleted experiment successfully:
```
import mlflow
from mlflow.tracking.client import MlflowClient

mlflow.set_experiment("DogExperiment")
client = MlflowClient()
exp = client.get_experiment("1")
print(exp)

client.delete_experiment("1")

client.set_experiment_tag("1", "DOG", "CAT")

exp = client.get_experiment("1")
print(exp)
```

Log output:
```
$ python filesetinactive.py
INFO: 'DogExperiment' does not exist. Creating a new experiment
<Experiment: artifact_location='file:///Users/czumar/mlflow/DEBUG/exptag/mlruns/1', experiment_id='1', lifecycle_stage='active', name='DogExperiment', tags={}>
<Experiment: artifact_location='file:///Users/czumar/mlflow/DEBUG/exptag/mlruns/1', experiment_id='1', lifecycle_stage='deleted', name='DogExperiment', tags={'DOG': u'CAT'}>
```

We should fix this and add a test to ensure that an exception is thrown instead.

Reply:
Yeah I think I got lost in the validation logic - my bad! Fixed and added a test.